packages/x-date-pickers-pro/src/StaticDateRangePicker/StaticDateRangePicker.tsx

```diff
@@ -246,6 +246,8 @@ StaticDateRangePicker.propTypes = {
   renderLoading: PropTypes.func,
   /**
    * Disable specific date.
+   *
+   * Warning: This validation will run on every rendered date. Expensive computations can impact performance.
```

Comment:
"Every rendered date" might not be super accurate because it's not true on the month / year views, we call it on every day of the current month (in the day view)
We call it for the selected value (can be called 2-3 times depending on the context)

```suggestion
   * Warning: This function can be called dozens of time (e.g: at least once per day in the `day` view). Expensive computations can impact performance.
```

Reply:
IMHO the existing change seems sufficient. It's concise enough, while also being correct (it will be run only on rendered days). 🤔

Reply:
It is not run "on every rendered date"

Reply:
Oh, you mean that there are also instances in the `DateCalendar` lifecycle where this callback is also being run (like when moving the focused date)?
Then yeah, maybe focusing less on the initial render cost makes sense. 👌 
```suggestion
   * Warning: This function can be called multiple times (e.g. when rendering date calendar, checking if focus can be moved to a certain date, etc.). Expensive computations can impact performance.
```